syntax = "proto3";

package mypackage;

// Service LoadBalancerService pour la gestion de l'allocation des VMs
service LoadBalancerService {
    rpc GetPCResources (Empty) returns (PCList);
    rpc AllocateVMToPC (VMAllocationRequest) returns (VMAllocationResponse);
}

// Message pour récupérer les informations sur les PCs
message PCResources {
    string pc_id = 1;
    float cpu_usage = 2;
    float ram_usage = 3;
    string status = 4;
}

message PCList {
    repeated PCResources pcs = 1;
}

// Message pour allouer une VM sur un PC spécifique
message VMAllocationRequest {
    string vm_id = 1;
    string template_id = 2;
}

message VMAllocationResponse {
    string pc_id = 1;
    string message = 2;
}
